### PR TITLE
[openimageio] Bump to 2.1.11.2

### PR DESCRIPTION
**Pull request by theblackunknown**
This pull requests bump OpenImageIO port version to 2.1.10

- What does your PR fix? N/A

- Which triplets are supported/not supported? Have you updated the CI baseline?
I have tested a custom triplet x64-windows-static-release, but I am waiting on CI to iterate and validate others
I haven't touched the CI baseline

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

### CODE REVIEW

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
Hi @theblackunknown thanks for this PR.
Could you update the requested changes?

**Comment by theblackunknown** [comment]
Now that I am aware of #8543 from @JackBoosY I would suggest to put this one on hold 
I'll rebase mine on top of Jack's one once merged

File: ports/openimageio/portfile.cmake [hunk 1]
```diff
@@ -3,18 +3,13 @@ include(vcpkg_common_functions)
 vcpkg_from_github(
```
[the remark this answers is not in the document]
Comment (theblackunknown) [reply ports/openimageio/portfile.cmake]: Since which version of vcpkg this include is no more needed ?

[the remark this answers is not in the document]
Comment (theblackunknown) [reply ports/openimageio/portfile.cmake]: Thanks for the reference, it was not obvious from the changelog but I see it now
I'll update the review when I go back to my laptop

**Comment by huargovfx** [comment]
Good morning, version 2.1.11 released. I was having issues using this latest version with the port. It seems the way the patches have to be applied have changed from 2.1.10 to 2.1.11.

**Comment by theblackunknown** [comment]
Hi @huargovfx , I consider this PR on hold for now
The idea is to rebase my changes - and ideally upgrade to latest OpenImageIO - once #8543 is merged

If possible I would suggest to use this other PR is you don't need the latest OpenImageIO, otherwise please be a bit more patient :)

**Comment by JackBoosY** [comment]
@theblackunknown My PR #8543 has been merged, you can continue your work now : )

**Comment by theblackunknown** [comment]
Thanks ! Will do once I rebase my work !

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
Need to test features.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
Hi @theblackunknown 

Sorry for the late reply.

All features except for `field3d`(only supports Windows) passed with the following triplets:
- x86-windows
- x64-windows
- x64-windows-static

All features except for(`libraw`, `pybind11`) passed with x64-linux triplet.

Could you please help confirm and check the results about `libraw `and `pybind11` on Linux?

**Comment by theblackunknown** [comment]
Hey @NancyLi1013 , I have just tested those features on a Linux OS
I have not encountered any trouble for `libraw`, however I did for `pybind11`.

In my opinion, the main issue is how `pybind11` port manage its dependencies :
- On Windows, it relies on `python3` from vcpkg port : this provide a consisten, controlled and reproducable behavior
- On other OSes, it relies on whatever system python is present : this is very inconsistent, and error prone when trying to have reproducable builds on different installations

Now, even if I found the correct package to install to make it work on Linux, I had to further tweak the portfile to make sure `openimageio` will use the python shared library and not the static one because otherwise the port will fail (because the former is compiled with -fPIC which is what openimageio python bindings requires)

So my last commit is hacky but make it pass for me,
I would suggest to fix the following ports :
- `python3` by building shared library (currently tackled in #9979) 
- `pybind11` by always relying on the upgraded `python3` to get consistent and reproducable builds over any installations

What do you think ?

**Comment by JackBoosY** [comment]
@theblackunknown 
I think we must guarantee to use the built-in tools (git, python ...) of vcpkg to ensure that the actual effect in each triplet is consistent. And update these tools to make changes in all triplets at the same time.

**Comment by theblackunknown** [comment]
@JackBoosY 
Just to be sure to understand, in the current case what are you suggesting to fix the issue ?
Make pybind11 relies on vcpkg port ? vcpkg downloaded python ? or something else ?

**Comment by JackBoosY** [comment]
@theblackunknown 
- If OIIO depends on pybind, use vcpkg port `pybind11` (Add `Build-Depends: pybind11` to `CONTROL`)
- If OIIO use python3, use vcpkg installed tools python3(Use `vcpkg_find_acquire_program`)

**Comment by theblackunknown** [comment]
Ok, we agree on this and this is what the current port is currently doing

However the pybind11 port needs an upgrade because the wrong python libraries gets picked up in its current state :
- pybind11 relies on system python instead of vcpkg python
- we need to have python with shared library

I am currently tackling this issue in #9979

**Comment by theblackunknown** [comment]
@NancyLi1013 @JackBoosY same as my other PR, this one is also on hold as it requires the ones that are now closed
I am closing it for now until I go back to it

**Pull request closed without merging**